solutions/silver/ccc-Firehose.mdx

```diff
@@ -87,4 +87,148 @@ int main() {
 ```
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Firehose {
+    static final int MAX_H = 1000;
+    static final int STREET_SIZE = 1000000;
+
+    static boolean possible(int length, int numHouses, int numHydrants, int[] houses) {
+        for (int i = 0; i < numHouses; i++) {
+
+            // Number of fire hydrants that are needed.
+            int needed = 0;
+
+            // The house that we need to connect a fire hydrant to.
+            int start = houses[i];
+
+            for (int j = 1; j < numHouses; j++) {
+                // Address of house at index j.
+                int end = houses[(i + j) % numHouses];
+
+                /* 
+                 * If the distance between the start and end houses is greater than
+                 * two times the length of the hose, a single fire hydrant will not
+                 * be enough to cover both houses.
+                 */
+                int dist = (end - start + STREET_SIZE) % STREET_SIZE;
+                if (dist > 2 * length) {
+                    start = end;
+                    needed++;
+                }
+            }
+
+            // Increment needed as we need another hydrant to reach the last houses.
+            needed++;
+
+            if (needed <= numHydrants) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static void main(String[] args) {
+        Kattio io = new Kattio();
+        
+        int numHouses = io.nextInt();
+        int[] houses = new int[numHouses];
+        
+        for (int i = 0; i < numHouses; i++) {
+            houses[i] = io.nextInt();
+        }
+        
+        int numHydrants = io.nextInt();
+        
+        Arrays.sort(houses);
+
+        // Binary search for the lowest hose length
+        int left = 0, right = STREET_SIZE, ans = 0;
```

Comment:
declarations on separate lines

Reply:
for all three

Reply:
ye

Reply:
done

Reply:
How foolish of me! I thought I was making changes in the Java section but didn't realise I was keeping making changes in the C++ section.😂